src/Exception/MiddlewareNotCallableException.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Mvc\Exception;
+
+final class MiddlewareNotCallableException extends RuntimeException
```

Comment:
Since we're allowing either callable or http-interop middleware, perhaps we should change this to `InvalidMiddlewareException` instead?